routes/helpers/render.js

```diff
@@ -62,6 +62,14 @@ let render = {
 
     vars.urlPath = url.parse(req.originalUrl).pathname;
 
+    vars.returnTo = '/';
+    if (req.query.returnTo) {
+        vars.returnTo = req.query.returnTo;
+    } else if (req.path == 'signin' || req.path == 'register') { // do nothing
```

Comment:
This condition won't be true, since the path will be /signin or /register (note leading slash), so you will instead move on to the `else` block. That's why right now http://localhost/signin (without returnTo) still redirects to itself. 

In addition to fixing that check, any reason not to move that initial `vars.returnTo = '/'` inside this `else if` block?

Reply:
You're correct, I only tested it with querys present.

No, just how my brain was working that day.  I'll move it, the logic is readable if it's inside the block.